index.ts

```diff
@@ -0,0 +1,403 @@
+import { createMachine, createActor, assign, raise } from "xstate";
+import { speechstate, Settings, Hypothesis } from "speechstate";
+
+const azureCredentials = {
+  endpoint:
+    "https://northeurope.api.cognitive.microsoft.com/sts/v1.0/issuetoken",
+  key: "994d7aefd0894deca17e2b24dc6b7daf",
+};
+
+
+const settings: Settings = {
+  azureCredentials: azureCredentials,
+  asrDefaultCompleteTimeout: 0,
+  locale: "en-US",
+  asrDefaultNoInputTimeout: 5000,
+  ttsDefaultVoice: "en-GB-RyanNeural",
+};
+// air-conditioner functions 
+interface DMContext {
+  spstRef?: any;
+  lastResult?: Hypothesis[];
+  Mode?: string; // warm or cool
+  Switch?: string ; // on or off
+  Intensity?: string; // weak, strong, or medium
+  Temperature?: string; // high or low 
+  Direction?: string; // body,legs or loop (which is the whole car)
+  userInput?:string;
+  
+}; 
+
+// helper functions
+const say =
+  (text: string) =>
+  ({ context }) => {
+    context.spstRef.send({
+      type: "SPEAK",
+      value: { utterance: text },
+    });
+  };
+const listen =
+  () =>
+  ({ context }) =>
+    context.spstRef.send({
+      type: "LISTEN",
+    });
+
+interface Grammar {
+    [index: string]: {
+      intent: string;
+      entities: {
+        [index: string]: string;
+      };
+    };
+  }
+
+// I changed the grammar a bit, as well as the utterances what is being said by the system 
+// it's stark with saying which function one can choose from, and strats then with "Let's start with mode", there one can say "cool", or warm
+// then it goes direction there one can say "legs" or "body", for the temperature one can say "high" or "low", and lastly for the intensity
+// one can say "strong", "medium", or "weak". 
```

Comment:
I see that you support only basic requests and one can only provide information slot by slot. According to the task, you should support one-shot request (e.g. "Make AC strong and hot and point it towards my legs") as well as over-answering, for instance.
```
U> Make AC hot
S> Ok, AC now is in hot mode. Which direction?
U> Towards my legs and very strong, please. 
```